meta-mbl-distro/scripts/create-update-payload

```diff
@@ -239,108 +286,131 @@ def get_files_for_boot_part(deploy_dir, tinfoil=None):
     return boot_files
 
 
-def copy_files_to_subdir(paths, dst_dir):
+def get_archived_file_spec_for_rootfs(image_name, tinfoil):
+    """
+    Get an ArchivedFileSpec for the root partition image.
     """
-    Copy files to a destination directory, renaming them when specified.
+    deploy_dir = get_bitbake_conf_var("DEPLOY_DIR_IMAGE", tinfoil)
+    if not deploy_dir:
+        bb.fatal("Failed to get value of DEPLOY_DIR_IMAGE variable")
 
-    :param paths list<Path|tuple<Path, Path>>: a list of specifications of
-    files to copy to a destination directory. Each element can just be a
-    plain Path, in which case it is copied to the destination directory with
-    its existing filename; otherwise the element should be a 2-tuple of Paths
-    (src_path, dst_name) where src_path will be copied to dst_dir with the name
-    dst_name.
+    machine = get_bitbake_conf_var("MACHINE", tinfoil)
+    if not machine:
+        bb.fatal("Failed to get value of MACHINE variable")
 
-    :param dst_dir Path: the directory into which files will be copied.
-    """
-    for path in paths:
-        if isinstance(path, tuple):
-            src_path = path[0]
-            dst_name = path[1].name
-        else:
-            src_path = path
-            dst_name = path.name
-        # copy2 preserves file metadata and follows symlinks
-        try:
-            shutil.copy2(str(src_path), str(dst_dir / dst_name))
-        except FileNotFoundError:
-            bb.fatal(
-                "File for the {} update component was not found at path {}. "
-                "Have you built the update components?".format(
-                    dst_name, str(src_path)
-                )
+    rootfs_filename = "{}-{}.tar.xz".format(image_name, machine)
+    try:
+        rootfs_image_path = pathlib.Path(deploy_dir, rootfs_filename).resolve()
+    except FileNotFoundError:
+        bb.fatal(
+            "The rootfs image with filename {} was not found. Have you "
+            "provided the correct image-name? Have you built the rootfs "
+            "image and is it located in {}?".format(
+                rootfs_filename, deploy_dir
             )
+        )
 
+    if not rootfs_image_path.exists():
```

Comment:
Oh, we're creating the resolved `Path` because we check if the file exists here? Maybe we should actually move this check into the `ArchivedFileSpec::__init___` method? It would make the calling code cleaner and it's probably good to know if any of these archive files don't exist.

Reply:
Done in https://github.com/ARMmbed/meta-mbl/pull/730/commits/550dd85fe34f39728ae8110a922ff15d5eadc2a0.